contracts/utils/ReentrancyGuard.sol

```diff
@@ -61,6 +61,25 @@ abstract contract ReentrancyGuard {
         _nonReentrantAfter();
     }
 
+    /**
+     * @dev A `view` only version of {nonReentrant}. Use to block view functions
+     * from being called, preventing reading from inconsistent contract state.
+     *
+     * CAUTION: This is a "view" modifier and does not change the reentrancy
+     * status. Use it only on view functions. For payable or non-payable functions,
+     * use the standard {nonReentrant} modifier instead.
+     */
+    modifier nonReentrantView() {
+        _nonReentrantBeforeView();
+        _;
+    }
+
+    function _nonReentrantBeforeView() private view {
+        if (_status == ENTERED) {
+            revert ReentrancyGuardReentrantCall();
+        }
+    }
```

Comment:
Would it make sense to reuse `_nonReentrantBeforeView` in `_nonReentrantBefore`?


```diff
function _nonReentrantBefore() private {
-  if (_status == ENTERED) {
-     revert ReentrancyGuardReentrantCall();
-  }
+  _nonReentrantBeforeView();
}
```

Same for ReentrancyGuardTransient